Source/WebCore/html/CanvasBase.cpp

```diff
@@ -411,6 +609,8 @@ bool CanvasBase::postProcessPixelBufferResults(Ref<PixelBuffer>&& pixelBuffer, c
     HashMap<uint32_t, uint32_t> pixelColorMap;
     bool wasPixelBufferModified { false };
 
+    HashMap<size_t, std::pair<size_t, size_t>> gradientIndiceMap;
```

Comment:
I don't quite understand this map.
If I have 2000x2000 canvas with a full-screen gradient, does this map get 4 million entries?

Reply:
I'm sure a fast algorithm to add visually non-distracting noise to a picture has been a solved research question for a while now. Maybe we could just look up a proper scientific noise kernel from some research paper and implement that?